### PR TITLE
Parallel reports

### PR DESCRIPTION
**Pull request by BlueBasilisk**
This PR contains:
- Minor YAML change: move number_of_processes in ExploratoryAnalysis instruction definition one level up; outside of 'analyses'
- make number_of_processes available to Reports, allowing for parallelization
- HTML readability: Added 'info' to ReportResult, describing the report in one or few sentences. Display this in the generated HTML. Also display generated 'output text' in HTML (prior we only displayed figures and tables), always show the report name (key) in the HTML, and minor other refactoring for general HTML readability

### CODE REVIEW

**Review by pavlovicmilena — COMMENTED** [review]
One question: if the parent report classes were added a "number_of_processes" parameter, shouldn't it also be added to all subclasses that are actually invoked? :) otherwise looks good :)

**Comment by BlueBasilisk** [comment]
Will merge first with another PR (replacing label strings wiht label objects in MLMethods). Didn't see your comments earlier, will also address those in the new PR.

File: immuneML/reports/encoding_reports/DesignMatrixExporter.py [hunk 1]
```diff
@@ -61,7 +61,9 @@ def _generate(self) -> ReportResult:
         details_result = self._export_details()
         label_result = self._export_labels()
 
-        return ReportResult(self.name, output_tables=[matrix_result, label_result], output_text=[details_result])
+        return ReportResult(self.name,
+                            info="The design matrix and related information of a given encoded Dataset",
```
[the remark this answers is not in the document]
Comment (BlueBasilisk) [reply immuneML/reports/encoding_reports/DesignMatrixExporter.py:65]: I agree, although in some reports the descriptions would be made using some dynamic information like user-defined parameters

File: immuneML/reports/encoding_reports/FeatureComparison.py [hunk 2]
```diff
@@ -87,6 +87,7 @@ def __init__(self, dataset: Dataset = None, result_path: Path = None, comparison
         self.keep_fraction = keep_fraction
         self.result_name = "feature_comparison"
         self.name = name
+        self.info = "Compares the feature values in a given encoded data matrix across two values for a metadata label. Each point in the resulting scatterplot represents one feature, and the values on the x and y axes are the average feature values across examples of two different classes. "
```
[the remark this answers is not in the document]
Comment (BlueBasilisk) [reply immuneML/reports/encoding_reports/FeatureComparison.py:90]: what did you mean by "and it should be the same for all reports"?

**Pull request closed without merging**